### DISCUSSION TITLE
Add support for passing an ng-template reference in the [template]

### DISCUSSION THREAD

**Discussion by ozkoidi** [Napster2210/ngx-spinner#266] [Ideas]
It would be useful to be able to pass the reference of an ng-template to the [template] property. This would allow us to have something like this 

```
<ngx-spinner [template]="loadingTemplate"></ngx-spinner>

<ng-template #loadingTemplate>
  <div>
    <span>Uploading...</span>
    <progress [value]="uploadProgress"> {{ uploadProgress }}% </progress>
  </div>
</ng-template>
```

I think this feature was already proposed on some issues and other Angular libraries such as [ngx-loading](https://github.com/Zak-C/ngx-loading) already support it.

**Comment by Napster2210** [comment]
This is resolved already.
